internal/provider/waypoint/resource_waypoint_application_template.go

```diff
@@ -214,6 +242,22 @@ func (r *ApplicationTemplateResource) Create(ctx context.Context, req resource.C
 		)
 	}
 
+	varOpts := []*waypoint_models.HashicorpCloudWaypointTFModuleVariable{}
+	for _, v := range plan.TerraformVariableOptions {
+		strOpts := []string{}
+		diags = v.Options.ElementsAs(ctx, &strOpts, false)
+		if diags.HasError() {
+			return
+		}
+
+		varOpts = append(varOpts, &waypoint_models.HashicorpCloudWaypointTFModuleVariable{
+			Name:         v.Name.ValueString(),
+			VariableType: v.VariableType.ValueString(),
+			Options:      strOpts,
+			UserEditable: false,
```

Comment:
Do we want to allow `UserEditable` to be set?

Reply:
Yes we do, I will add that as a `bool` field.